src/TextView.js

```diff
@@ -0,0 +1,155 @@
+import * as Notification from './utils/Notification.js';
+
+/** @module TextView */
+
+const $ = require('jquery');
+
+/*
+ * Class that manages getting the text for syllables in Neon from the mei file
+ */
+class TextView {
+	/**
+	 * A constructor for a TextView.
+	 * @param {NeonView} neonView = The NeonView parent.
+	 */
+	constructor(neonView) {
+		this.neonView = neonView;
+		let notificationSent = false;
+		// add checkbox to enable/disable the view
+		let block = document.getElementById('extensible-block');
+		let label = document.createElement('label');
+		let input = document.createElement('input');
+		label.classList.add('checkbox');
+		label.textContent = 'Display Text: ';
+		input.classList.add('checkbox');
+		input.id = 'displayText';
+		input.type = 'checkbox';
+		input.checked = false;
+		label.appendChild(input);
+		block.prepend(label);
+
+		$('#edit_mode').on('click', () => {
+			this.setTextEdit();
+		});
+		this.setTextViewControls();
+		this.neonView.view.addUpdateCallback(this.updateTextViewVisibility.bind(this));
+	}
+
+	/**
+ 	* set listeners on textview visibility checkbox
+ 	*/
+	setTextViewControls () {
+		this.updateTextViewVisibility();
+		$('#displayText').on('click', () => {
+			this.updateTextViewVisibility();
+		});
+	}
+
+	/*
+ 	 * set text to edit mode
+	 */
+	setTextEdit() {
+		let spans = Array.from($('#syl_text').children('p').children('span'));
+		spans.forEach(span => {
+			$(span).off('click');
+   			$(span).on('click', () => {
+   				this.updateSylText(span);
+   			});
+		});
+	}
+
+	/*
+ 	* update the text for a single syl element
```

Comment:
Replace with jsdoc comment and add parameter field (`@param {HTMLElement} span`)